src/vcModals.cpp

```diff
@@ -376,34 +368,45 @@ void vcModals_DrawWelcome(vcState *pProgramState)
       ImGui::Spacing();
       ImGui::Spacing();
 
-      UDCOMPILEASSERT(udLengthOf(pNewProjectTypes) == udLengthOf(projectIcons), "Invalid matching sizes");
-
-      for (size_t i = 0; i < udLengthOf(pNewProjectTypes); ++i)
+      for (size_t i = 0; i < udLengthOf(items); ++i)
       {
         bool selected = false;
         if (ImGui::Selectable(udTempStr("##newProjectType%zu", i), &selected, ImGuiSelectableFlags_DontClosePopups, ImVec2(475, 48)))
         {
-          creatingNewProjectType = (int)i;
-          udStrcpy(pProgramState->modelPath, vcString::Get("modalProjectNewTitle"));
+          if (items[i].icon == vcMBBI_Geospatial || items[i].icon == vcMBBI_Grid || items[i].icon == vcMBBI_ExpertGrid)
+          {
+            creatingNewProjectType = (int)i;
+            udStrcpy(pProgramState->modelPath, vcString::Get("modalProjectNewTitle"));
 
-          if (i == 0) // Geolocated
-            zoneCustomSRID = 84;
-          else if (i == 1)
-            zoneCustomSRID = 0; // Non Geolocated
+            if (items[i].icon == vcMBBI_Geospatial)
+              zoneCustomSRID = 84; // Geolocated
+            else if (items[i].icon == vcMBBI_Grid)
+              zoneCustomSRID = 0; // Non Geolocated
```

Comment:
Is `0` standard for non-geolocated? I wonder if we should use an enum for common srid values.